lib/actions/action.js

```diff
@@ -0,0 +1,28 @@
+class Action {
+  /**
+   * @param eventName
+   *  An event name to be evaluated for support. The name is as in the GitHub
+   *  webhook format of issues.opened, pull_request.opened, etc
+   *
+   * @return boolean true if the validator supports the event. i.e. issues.opened
+   */
+  isEventSupported (eventName) {
+    return isEventInContext(eventName, this.supportedEvents)
+  }
+
```

Comment:
I recommend declaring an `async` method here called `execute` -- which is the common interface for all actions to be executed by the `Mergeable` / executor. In this implementation it would throw an error. Here's why:

There's are no interfaces in JS but the caller needs a common interface to call, in this case it's `execute` -- if for any reason a child class forgot to override this method correctly then at least the system can return appropriate errors as opposed to unknown errors.

Reply:
i have a different methods for action than validators, i have `doPreAction` and `doPostAction`, this is for actions like checks which may require we do something before we run validation, idk how execute will fit into this

Reply:
Based on the names of those methods. `doPreAction` is called before the "action" method is invoked and `doPostAction` is called after the "action" method is invoked -- Is that correct? What is the method that is in between the two? Or is there none?

Reply:
hmm.. i think the better names would be `doPreValidationAction`, `doPostValidationAction`. for example, in checks, preValidationAction would be to create a pending status, post would update it to completed status

Reply:
Action is something that executes after validation. The main method for execution is after validation is done. I fine it totally OK to name it `execute` and having a `beforeValidation` method.

Alternatively If you want to be more explicit which might be better -- you can name them `beforeValidation` and `afterValidation`. Both are familiar naming conventions in frameworks.

Reply:
I do recommend to declare them in this base class and have them throw Error.